### PR TITLE
Testing part 2

### PR DESCRIPTION
**Pull request by rmjspecial**
**Project Description**

- created a function for editing the task description.
- Add a jest library.
- Add mock to update local storage

### CODE REVIEW

File: src/services.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+export function createLocalStorage(key = 'tasks', value = []) {
+  if (!getData(key)) {
+    localStorage.setItem(key, JSON.stringify(value));
+  }
+}
+
+export function updateTasks(data) {
+  localStorage.setItem('tasks', JSON.stringify(data));
+}
```
Comment (gtekle) [inline src/services.js:19]: - [ ] Kindly add unit test for this function as it is part of this milestone's requirements. 
image

File: src/update_complete.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export function clearCompleteTasks() {
+  let tasks = getData('tasks');
+
+  tasks = tasks.filter((task) => !task.completed);
+
+  updateTasks(tasks);
+  updateIndexes();
+}
```
Comment (gtekle) [inline src/update_complete.js:29]: - [ ] Kindly add unit test for this function as it is part of this milestone's requirements.
image

File: src/update_complete.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  tasks = tasks.map((task) => {
+    if (id === task.index) {
+      return { ...task, completed: isCompleted };
+    }
+
+    return task;
+  });
+
+  updateTasks(tasks);
+}
```
Comment (gtekle) [inline src/update_complete.js:20]: - [ ] Kindly add unit test for this function as it is part of this milestone's requirements.
image

**Review by gtekle — CHANGES_REQUESTED** [review]
Hi @chimwemwe007 :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost-there-01

## Highlights
:heavy_check_mark: Professional ReadMe
:heavy_check_mark: Meaningful commit messages
:heavy_check_mark: Descriptive PR title and summary

## Required Changes ♻️
_Check the comments under the review._

- [x] You are directly accessing the localStoarge in your test. Thus, kindly mock the localStorage   as it is this milestone's requirement and mocking external application makes your tests to run faster.
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gtekle** [comment]
Hi @chimwemwe007 
As per our discussion over slack, since you are using Jest version 28.1.3 you can disregard my previous suggestion to mock localStorage.

Happy coding 👍

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @rmjspecial  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**